lib/BC/SleighLifter.cpp

```diff
@@ -572,9 +573,10 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock {
       }
       case OpCode::CPUI_FLOAT_INT2FLOAT: {
         auto int2float_inval = this->LiftIntegerInParam(bldr, input_var);
-        if (int2float_inval.has_value()) {
-          auto *converted = bldr.CreateSIToFP(
-              *int2float_inval, llvm::Type::getFloatTy(this->context));
+        auto new_float_type = this->GetFloatTypeOfByteSize(outvar->size);
+        if (int2float_inval.has_value() && new_float_type) {
+          auto *converted =
```

Comment:
```suggestion
          auto converted =
```
(I know it is not change that happened in this PR, but since I see it in the diff)